apps/ledger-live-desktop/src/renderer/components/WebPlatformPlayer/LiveAppSDKLogic.ts

```diff
@@ -0,0 +1,391 @@
+  accountId: string,
+  transaction: RawPlatformTransaction,
+  params: any,
+) => {
+  tracking.platformSignTransactionRequested(manifest);
+
+  const platformTransaction = deserializePlatformTransaction(transaction);
+
+  const account = accounts.find(account => account.id === accountId);
+
+  if (!account) {
+    tracking.platformSignTransactionFail(manifest);
+    return Promise.reject(new Error("Account required"));
+  }
+
+  const parentAccount = isTokenAccount(account) ? getParentAccount(account, accounts) : null;
+
+  if (
+    (isTokenAccount(account) ? parentAccount?.currency.family : account.currency.family) !==
+    platformTransaction.family
+  ) {
+    throw new Error("Transaction family not matching account currency family");
+  }
+
+  const { canEditFees, liveTx, hasFeesProvided } = getPlatformTransactionSignFlowInfos(
+    platformTransaction,
+  );
+
+  return new Promise((resolve, reject) =>
+    dispatch(
+      openModal("MODAL_SIGN_TRANSACTION", {
+        canEditFees,
+        stepId: canEditFees && !hasFeesProvided ? "amount" : "summary",
+        transactionData: liveTx,
+        useApp: params.useApp,
+        account,
+        parentAccount,
+        onResult: (signedOperation: SignedOperation) => {
+          tracking.platformSignTransactionSuccess(manifest);
+          resolve(serializePlatformSignedTransaction(signedOperation));
+        },
+        onCancel: (error: Error) => {
+          tracking.platformSignTransactionFail(manifest);
+          reject(error);
+        },
+      }),
+    ),
+  );
+};
+
+export const broadcastTransactionLogic = async (
+  { manifest, dispatch, accounts }: WebPlatformContext,
+  accountId: string,
+  signedTransaction: RawPlatformSignedTransaction,
+  pushToast: (data: ToastData) => void,
+  t: TFunction,
+) => {
+  const account = accounts.find(account => account.id === accountId);
+  if (!account) {
+    tracking.platformBroadcastFail(manifest);
+    return Promise.reject(new Error("Account required"));
+  }
+
+  if (!signedTransaction) {
+    tracking.platformBroadcastFail(manifest);
+    return Promise.reject(new Error("Transaction required"));
+  }
+
+  const parentAccount = isTokenAccount(account) ? getParentAccount(account, accounts) : null;
+
+  const signedOperation = deserializePlatformSignedTransaction(signedTransaction, accountId);
+  const bridge = getAccountBridge(account, parentAccount);
+  const mainAccount = getMainAccount(account, parentAccount);
+
+  let optimisticOperation: Operation = signedOperation.operation;
+
+  // FIXME: couldn't we use `useBroadcast` here?
+  if (!getEnv("DISABLE_TRANSACTION_BROADCAST")) {
+    try {
+      optimisticOperation = await bridge.broadcast({
+        account: mainAccount,
+        signedOperation,
+      });
+      tracking.platformBroadcastSuccess(manifest);
+    } catch (error) {
+      tracking.platformBroadcastFail(manifest);
+      throw error;
+    }
+  }
+
+  dispatch(
+    updateAccountWithUpdater(account.id, account =>
```

Comment:
I can also merge my fix so that you can rebase on it and you don't have to also change the modal code
```suggestion
    updateAccountWithUpdater(mainAccount.id, account =>
```

Reply:
I took your code suggestion